test/ex_sdp/connection_data_test.exs

```diff
@@ -2,135 +2,123 @@ defmodule ExSDP.ConnectionDataTest do
+               address_count: nil,
+               address: {224, 2, 1, 1},
+               network_type: "IN"
              }
     end
   end
 
   describe "Connection information parser when working with ip6" do
     test "parses valid connection with count param" do
-      assert {:ok, connections} = ConnectionData.parse("IN IP6 FF15::101/3")
-
-      assert connections == [
-               %IP6{
-                 value: {65_301, 0, 0, 0, 0, 0, 0, 257}
-               },
-               %IP6{
-                 value: {65_301, 0, 0, 0, 0, 0, 0, 258}
-               },
-               %IP6{
-                 value: {65_301, 0, 0, 0, 0, 0, 0, 259}
+      assert {:ok, connection_data} = ConnectionData.parse("IN IP6 FF15::101/3")
+
+      assert connection_data ==
+               %ConnectionData{
+                 address: {65_301, 0, 0, 0, 0, 0, 0, 257},
+                 address_count: 3,
+                 ttl: nil,
+                 network_type: "IN"
                }
-             ]
     end
 
     test "parses valid connection" do
-      assert {:ok, connections} = ConnectionData.parse("IN IP6 FF15::103")
+      assert {:ok, connection_data} = ConnectionData.parse("IN IP6 FF15::103")
 
-      assert connections == %IP6{
-               value: {65_301, 0, 0, 0, 0, 0, 0, 259}
+      assert connection_data == %ConnectionData{
+               address: {65_301, 0, 0, 0, 0, 0, 0, 259},
+               address_count: nil,
+               ttl: nil,
+               network_type: "IN"
              }
     end
   end
 
   describe "Connection information parser returns an error when" do
     test "connection spec is invalid" do
-      assert {:error, :invalid_connection_data} = ConnectionData.parse("IN EPI")
+      assert {:error, {:invalid_connection_data, :too_few_fields}} =
+               ConnectionData.parse("IN EPI")
     end
 
     test "address is not valid" do
-      assert {:error, :invalid_address} = ConnectionData.parse("IN IP4 224.2.1.1/127/3/4")
+      assert {:error, {:invalid_connection_data, :invalid_ttl_or_address_count}} =
+               ConnectionData.parse("IN IP4 224.2.1.1/127/3/4")
     end
 
     test "either ttl or count is not an integer" do
-      assert {:error, :option_nan} = ConnectionData.parse("IN IP4 224.2.1.1/127/3d")
-      assert {:error, :option_nan} = ConnectionData.parse("IN IP4 224.2.1.1/127a/3")
-    end
+      assert {:error, {:invalid_connection_data, :invalid_ttl_or_address_count}} =
+               ConnectionData.parse("IN IP4 224.2.1.1/127/3d")
 
-    test "ttl is not in 0..255 range" do
-      assert {:error, :wrong_ttl} = ConnectionData.parse("IN IP4 224.2.1.1/256")
+      assert {:error, {:invalid_connection_data, :invalid_ttl_or_address_count}} =
+               ConnectionData.parse("IN IP4 224.2.1.1/127a/3")
     end
 
-    test "when address expansion overflows IP octet range" do
-      assert {:error, :invalid_address} = ConnectionData.parse("IN IP4 224.2.1.255/127/3")
```

Comment:
We no longer checks this. I couldn't find any requirements in RFC regarding max address number and not checking this simplifies code so I have removed it.